parsers/CA_PE.py

```diff
@@ -1,160 +1,145 @@
-        "pei_wind_used": _find_pei_key(raw_data, "Wind Power Used On Island"),
-        "pei_wind_exported": _find_pei_key(raw_data, "Wind Power Exported Off Island"),
-        "datetime": data_timestamp.datetime,
+    # Extract the timestamp.
+    timestamp = datetime.strptime(
+        event[-1]["data"]["text"][len("Last updated ") :], "%B %d, %Y %I:%M %p"
+    ).replace(tzinfo=TIMEZONE)
+    # Return a new, more convenient lookup table.
+    return {
+        "datetime": timestamp,
+        "fossil": _parse_power(modes["Total On-Island Fossil Fuel Generation"]),
+        "load": _parse_power(modes["Total On-Island Load"]),
+        "wind": _parse_power(modes["Total On-Island Wind Generation"]),
     }
 
-    # the following keys are always required downstream, if we don't have them, no sense returning
-    if data["pei_wind_gen"] is None or data["pei_fossil_gen"] is None:
-        return None
-
-    return data
-
 
 def fetch_production(
-    zone_key: str = "CA-PE",
+    zone_key: ZoneKey = ZONE_KEY,
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> dict[str, Any]:
+) -> list[dict[str, Any]]:
     """Requests the last known production mix (in MW) of a given country."""
+    if zone_key != ZONE_KEY:
+        raise ParserException("CA_PE.py", "Cannot parse zone '{zone_key}'", zone_key)
     if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
-
-    requests_obj = session or Session()
-    pei_info = _get_pei_info(requests_obj)
-
-    if pei_info is None:
-        return None
-
-    data = {
-        "datetime": pei_info["datetime"],
-        "zoneKey": zone_key,
-        "production": {
-            "wind": pei_info["pei_wind_gen"],
+        raise ParserException("CA_PE.py", "Unable to fetch historical data", ZONE_KEY)
+
+    event = _get_event(session or Session())
+    production_breakdowns = ProductionBreakdownList(logger)
+    production_breakdowns.append(
+        datetime=event["datetime"],
+        production=ProductionMix(
+            # Specify some sources that definitely aren't present on PEI as
+            # zero; this allows the analyzer to better estimate CO2eq.
+            coal=0,
+            geothermal=0,
+            hydro=0,
+            nuclear=0,
             # These are oil-fueled ("heavy fuel oil" and "diesel") generators
-            # used as peakers and back-up
-            "oil": pei_info["pei_fossil_gen"],
-            # specify some sources that definitely aren't present on PEI as zero,
-            # this allows the analyzer to better estimate CO2eq
-            "coal": 0,
-            "hydro": 0,
-            "nuclear": 0,
-            "geothermal": 0,
-        },
-        "storage": {},
-        "source": "princeedwardisland.ca",
-    }
-
-    return data
+            # used as peakers and back-up.
+            oil=event["fossil"],
+            wind=event["wind"],
+        ),
```

Comment:
Could we loop over these instead? I would prefer if we used the add_value() method to keep it in line with other parsers and make sure any current and future validation is run on it.